src/primitives/block.cpp

```diff
@@ -32,11 +32,13 @@ std::string CBlock::ToString() const
     return s.str();
 }
 
-size_t GetVirtualBlockSize(const CBlock& block)
-{
-    // The formula is: vsize = base_size + witness_size / 4.
-    // We can only serialize base or totalbase+witness, however, so the formula
-    // becomes: vsize = base_size + (total_size - base_size) / 4 or
-    // vsize = (total_size + 3 * base_size) / 4.
-    return (::GetSerializeSize(block, SER_NETWORK, 0) * 3 + ::GetSerializeSize(block, SER_NETWORK, SERIALIZE_TRANSACTION_WITNESS) + 3) / 4;
+size_t GetBlockCost(const CBlock& block)
+{    
+    // coreSize is the size of block without witness data
+    size_t coreSize = ::GetSerializeSize(block, SER_NETWORK, 0);
+    // witSize is the size of witness data: totalBlockSize - coreSize
+    size_t witSize = ::GetSerializeSize(block, SER_NETWORK, SERIALIZE_TRANSACTION_WITNESS) - coreSize;
+    assert(witSize >= 0);
```

Comment:
size_t is an unsigned integer.
do:
size_t witSize = ::GetSerializeSize(block, SER_NETWORK, SERIALIZE_TRANSACTION_WITNESS);
assert(witSize >= coreSize);
witSize -= coreSize;